src/main/java/org/elasticsearch/index/engine/internal/InternalEngine.java

```diff
@@ -1135,37 +1150,48 @@ public int compare(Segment o1, Segment o2) {
 
     @Override
     public void close() throws ElasticsearchException {
-        logger.debug("close now acquire writeLock");
-        try (InternalLock _ = writeLock.acquire()) {
-            logger.debug("close acquired writeLock");
-            if (isClosed.compareAndSet(false, true)) {
+        if (isClosed.get() == false) { // don't acquire the write lock if we are already closed
+            logger.debug("close now acquire writeLock");
```

Comment:
I think these should be trace (I think we added them in debug when debugging some build failures. We probably need to add a debug log after closeNoLock saying "engine closed"